### PR TITLE
VKAL-15587 Update Avi-init-secret

### PR DESCRIPTION
**Pull request by abhishekbsingh**

### CODE REVIEW

File: internal/lib/lib.go [hunk 1]
```diff
@@ -1601,7 +1601,11 @@ func RefreshAuthToken(kc kubernetes.Interface) {
 		return
 	}
 	token := newTokenResp.(map[string]interface{})["token"].(string)
-	aviSecret, err := GetAviSecretWithRetry(kc, retryCount)
+	secret := AviSecret
+	if utils.IsVCFCluster() {
+		secret = "avi-init-secret"
```
[the remark this answers is not in the document]
Comment (abhishekbsingh) [reply internal/lib/lib.go]: done

**Review by swathinsankaran — APPROVED** [review]
LGTM

File: internal/lib/dynamic_client.go [hunk 2]
```diff
@@ -313,13 +313,13 @@ func GetCNIPlugin() string {
 // is complicated business right now.
 func WaitForInitSecretRecreateAndReboot() {
 	cs := utils.GetInformers().ClientSet
-	if err := cs.CoreV1().Secrets("vmware-system-ako").Delete(context.TODO(), "avi-init-secret", metav1.DeleteOptions{}); err != nil {
+	if err := cs.CoreV1().Secrets("vmware-system-ako").Delete(context.TODO(), AviInitSecret, metav1.DeleteOptions{}); err != nil {
```
Comment (apalsule) [inline internal/lib/dynamic_client.go:316]: dont you need a nil check here ?

File: internal/lib/dynamic_client.go [hunk 3]
```diff
@@ -313,13 +313,13 @@ func GetCNIPlugin() string {
...
 		utils.AviLog.Errorf("Error while deleting the init Secret for Secret refresh.")
 		return
 	}
 
 	var checkForInitSecretRecreate = func(cs kubernetes.Interface) error {
-		_, err := cs.CoreV1().Secrets("vmware-system-ako").Get(context.TODO(), "avi-init-secret", metav1.GetOptions{})
+		_, err := cs.CoreV1().Secrets("vmware-system-ako").Get(context.TODO(), AviInitSecret, metav1.GetOptions{})
```
Comment (apalsule) [inline internal/lib/dynamic_client.go:322]: nil check ?

File: internal/lib/dynamic_client.go (hunk 2 above)
Comment (abhishekbsingh) [reply internal/lib/dynamic_client.go:316]: I did not understand, nil check for what? I don't think we need any nil check here

**Review by akshayhavile — APPROVED** [review]

**Pull request merged**